### PR TITLE
Migrate to JUnit 5 from JUnit 4 in related to `spring` module

### LINKED ISSUE #4788: Migrate JUnit 4 tests in Spring Boot integration modules to JUnit 5
Spring integration modules have to use JUnit 4 to share the test suites with the legacy Spring Boot 1.
https://github.com/line/armeria/blob/5bd1c7b08d783a8ea39fa2dd9259cec899ecb606/spring/boot2-autoconfigure/src/test/java/com/linecorp/armeria/spring/DeprecatedIpTest.java#L25-L42
Spring Boot 1 integration modules are going to be removed from Armeria. #4787

That means there is no reason we have to use JUnit 4 after #4787 is merged. It would be nice to use JUnit 5 like any other test code.

### PR DESCRIPTION
**Pull request by heowc**
Motivation:

Starting from spring boot 2.4.0, junit5 is recommended. That's why we need to respond in the same way.

Modifications:

- `org.junit.Test` -> `org.junit.jupiter.api.Test`
- Remove `org.junit.runner.RunWith`
- Change some `@AutoConfigureMetrics` to `@AutoConfigureObservability`
- `TestRule globalTimeout` -> `@Timeout(10)`

Result:

- Closes #4788

### CODE REVIEW

File: spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java [hunk 1]
```diff
@@ -54,17 +47,16 @@
...
+@Timeout(10)
+class ByteBufLeakTest {
 
     private static final AtomicInteger completed = new AtomicInteger();
     private static final Queue<NettyDataBuffer> allocatedBuffers = new ConcurrentLinkedQueue<>();
 
     private static final AtomicBoolean requestReceived = new AtomicBoolean();
 
     @SpringBootApplication
-    @Configuration
```
Comment (minwoox) [inline spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java:67]: So now `@Configuration` isn't needed anymore? If so, could you also remove this annotation in other tests?

**Review by minwoox — COMMENTED** [review]
Could you also remove `AutoConfigureMetrics` in the actuator test classes? 😉 
e.g. `ArmeriaSpringActuatorAutoConfigurationSecureTest`

**Comment by heowc** [comment]
> Could you also remove `AutoConfigureMetrics` in the actuator test classes? 😉 e.g. `ArmeriaSpringActuatorAutoConfigurationSecureTest`

I tried to change it, but those annotations have changed since 3.0.0, so I didn't modify the code to work with 2.x.

File: spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java (hunk 1 above)
Comment (heowc) [reply spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java:67]: `@SpringBootApplication` itself already contains `@Configuration`. So I thought it was enough to remove it.

`@SpringBootApplication`
   ㄴ `@SpringBootConfiguration`
       ㄴ `@Configuration`

Comment (minwoox) [reply spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java:67]: > So I thought it was enough to remove it.

Yes. How about also removing `@Configuration` in other classes? e.g.
https://github.com/line/armeria/pull/5036/files#diff-c1976a799fb6daafd188eecb992b2a31bb17d9147a5a1a324a565159b304fbb1R39
https://github.com/line/armeria/pull/5036/files#diff-27e240cb1631926a25e13a3c646e67aac657a603f3b4cf3be19b3a0e8e2e47faR60

**Comment by minwoox** [comment]
> I tried to change it, but those annotations have changed since 3.0.0, so I didn't modify the code to work with 2.x.

Ah, you changed it in the test of `it/spring/boo3-taomcat10`. I didn't notice it. Please forget about the comment. 😓

File: spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java (hunk 1 above)
Comment (heowc) [reply spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java:67]: Oh, I was missing something. I'll edit 😉

**Review by minwoox — APPROVED** [review]
Thanks a lot for cleaning this up, @heowc!

**Comment by ikhoon** [comment]
Q: Is there a reason that you didn't migrate the tests in the actuator modules?

**Comment by heowc** [comment]
> Q: Is there a reason that you didn't migrate the tests in the actuator modules?

`XXX-actuator` module seems to already work with junit5 🤔

File: spring/boot3-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaServerConfiguratorTest.java [hunk 2]
```diff
@@ -21,17 +21,16 @@
...
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.annotation.Order;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
 import com.linecorp.armeria.server.ServerBuilder;
 
-@RunWith(SpringJUnit4ClassRunner.class)
-public class ArmeriaServerConfiguratorTest {
+@SpringBootTest
```
Comment (jrhee17) [inline spring/boot3-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaServerConfiguratorTest.java:32]: nit; I don't think there's a need to load the entire context for this test

```suggestion
@ExtendWith(SpringExtension.class)
```

File: spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ArmeriaClientConfiguratorTest.java [hunk 3]
```diff
@@ -21,17 +21,16 @@
...
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.annotation.Order;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
 import com.linecorp.armeria.client.WebClientBuilder;
 
-@RunWith(SpringJUnit4ClassRunner.class)
-public class ArmeriaClientConfiguratorTest {
+@SpringBootTest
```
Comment (jrhee17) [inline spring/boot3-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ArmeriaClientConfiguratorTest.java:32]:
```suggestion
@ExtendWith(SpringExtension.class)
```

**Review by jrhee17 — COMMENTED** [review]
Thanks for the cleanup! Left a minor comment 👍

File: spring/boot3-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaServerConfiguratorTest.java (hunk 2 above)
Comment (heowc) [reply spring/boot3-autoconfigure/src/test/java/com/linecorp/armeria/spring/ArmeriaServerConfiguratorTest.java]: That's a good comment 👍

**Review by jrhee17 — APPROVED** [review]
Thanks for the cleanup 👍 🙇 👍

**Comment by ikhoon** [comment]
> > Q: Is there a reason that you didn't migrate the tests in the actuator modules?
> 
> `XXX-actuator` module seems to already work with junit5 🤔

Right. I was confused.

**Review by ikhoon — APPROVED** [review]
Thanks for cleaning up, @heowc! ❤️

**Review by trustin — APPROVED** [review]
Thanks a lot, @heowc! :bow:

**Pull request merged**